src/main/java/com/direwolf20/buildinggadgets/common/schema/template/TemplateIO.java

```diff
@@ -0,0 +1,21 @@
+package com.direwolf20.buildinggadgets.common.schema.template;
+
+import com.direwolf20.buildinggadgets.BuildingGadgets;
+import net.minecraft.nbt.CompressedStreamTools;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Optional;
+
+public final class TemplateIO {
+    public static Optional<byte[]> getCompressedBytes(Template template) {
```

Comment:
When else would this be used outside of the Template Packets? this seems way to specific to be in a generic class.

Reply:
As per Discord discussion, TemplateIO is not complete.